lib/web/apiserver.go

```diff
@@ -2050,9 +2050,11 @@ func (h *Handler) githubCallback(w http.ResponseWriter, r *http.Request, p httpr
 			logger.Debug("GitHub WebSession created with device web token")
 			// if a device web token is present, we must send the user to the device authorize page
 			// to upgrade the session.
-			// TODO (avatus) the web client currently doesn't handle any redirects after authorizing a web
-			// session with device trust. Once it does, append a redirect_url here as a query parameter
-			return fmt.Sprintf("/web/device/authorize/%s/%s", dwt.Id, dwt.Token)
+			redirectPath, err := BuildDeviceWebRedirectPath(dwt, res.ClientRedirectURL)
+			if err == nil {
+				return redirectPath
+			}
+			logger.WithError(err).Debug("Could not build redirect path")
```

Comment:
Is the intention to redirect to `res.ClientRedirectURL` anyway if `BuildDeviceWebRedirectPath` fails? I wonder if it'd help to be a little more explicit and do a normal `if err != nil` instead of depending on `return res.ClientRedirectURL` below – if the code gets moved or otherwise refactored, it'll be easier to keep the correct behavior.